### PR TITLE
SLCORE-629   Slim down TelemetryConstantAttributes and don't ask client for info we can infer on the backend

### PR DESCRIPTION
**Pull request by vojtech-suchy-sonarsource**
Don't ask about
- platform
- architecture
- NodeJs version

BugFix:  make sure to send constantAdditionalAttributes with liveAdditionalAttributes

### CODE REVIEW

File: core/src/main/java/org/sonarsource/sonarlint/core/NodeJsHelper.java [hunk 1]
```diff
@@ -61,8 +62,14 @@ public void detect(@Nullable Path configuredNodejsPath) {
     detectedNodePath = locateNode(configuredNodejsPath);
     if (detectedNodePath != null) {
       logOutput.log("Checking node version...", ClientLogOutput.Level.DEBUG);
-      var command = Command.create(detectedNodePath.toString()).addArgument("-v");
-      var nodeVersionStr = runSimpleCommand(command);
+      String nodeVersionStr;
+      var forcedNodeVersion = System.getProperty("sonarlint.internal.nodejs.forcedVersion");
+      if(Objects.nonNull(forcedNodeVersion)) {
```
Comment (henryju) [inline core/src/main/java/org/sonarsource/sonarlint/core/NodeJsHelper.java:67]:
```suggestion
      if (Objects.nonNull(forcedNodeVersion)) {
```

Comment (henryju) [reply core/src/main/java/org/sonarsource/sonarlint/core/NodeJsHelper.java:67]: Any reason to not simply use `forcedNodeVersion != null` ?

File: core/src/main/java/org/sonarsource/sonarlint/core/NodeJsHelper.java [hunk 2]
```diff
@@ -97,8 +104,11 @@ private Path locateNode(@Nullable Path configuredNodejsPath) {
     }
     logOutput.log("Looking for node in the PATH", ClientLogOutput.Level.DEBUG);
 
+    var forcedPath = System.getProperty("sonarlint.internal.nodejs.forcedPath");
     String result;
-    if (system2.isOsWindows()) {
+    if (Objects.nonNull(forcedPath)) {
```
Comment (henryju) [inline core/src/main/java/org/sonarsource/sonarlint/core/NodeJsHelper.java:109]: Any reason to not simply use `forcedPath != null` ?

File: core/src/main/java/org/sonarsource/sonarlint/core/telemetry/TelemetryServerAttributesProvider.java [hunk 3]
```diff
@@ -78,8 +90,16 @@ public TelemetryServerLiveAttributesDto getTelemetryServerLiveAttributes() {
...
     });
 
+    var nodeJsVersion = getNodeJsVersion();
+
     return new TelemetryServerLiveAttributesDto(usesConnectedMode, usesSonarCloud, devNotificationsDisabled, nonDefaultEnabledRules,
-      defaultDisabledRules);
+      defaultDisabledRules, nodeJsVersion);
+  }
+
+  @Nullable
```
Comment (henryju) [inline core/src/main/java/org/sonarsource/sonarlint/core/telemetry/TelemetryServerAttributesProvider.java:99]:
```suggestion
  @CheckForNull
```

File: core/src/main/java/org/sonarsource/sonarlint/core/telemetry/TelemetryServerAttributesProvider.java [hunk 4]
```diff
@@ -25,31 +25,43 @@
 import java.util.function.Predicate;
 import javax.inject.Named;
 import javax.inject.Singleton;
+import org.apache.commons.lang3.SystemUtils;
 import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
```
Comment (henryju) [inline core/src/main/java/org/sonarsource/sonarlint/core/telemetry/TelemetryServerAttributesProvider.java:30]: We should not use Jetbrains annotation in SLCORE, but the one from JSR 305

File: core/src/main/java/org/sonarsource/sonarlint/core/telemetry/TelemetryServerAttributesProvider.java [hunk 5]
```diff
@@ -25,31 +25,43 @@
...
     this.connectionConfigurationRepository = connectionConfigurationRepository;
     this.rulesService = rulesService;
     this.rulesRepository = rulesRepository;
+    this.nodeJsService = nodeJsService;
+  }
+
+  public TelemetryServerConstantAttributesDto getTelemetryServerConstantAttributes() {
+    var architecture = SystemUtils.OS_ARCH;
+    var platform = SystemUtils.OS_NAME;
+    return  new TelemetryServerConstantAttributesDto(platform, architecture);
```
Comment (henryju) [inline core/src/main/java/org/sonarsource/sonarlint/core/telemetry/TelemetryServerAttributesProvider.java:64]:
```suggestion
    return new TelemetryServerConstantAttributesDto(platform, architecture);
```

File: medium-tests/src/test/java/mediumtest/TelemetryMediumTests.java [hunk 6]
```diff
@@ -130,8 +135,9 @@ void it_should_create_telemetry_file_if_telemetry_enabled() {
         "  \"sonarlint_version\" : \"1.2.3\",\n" +
         "  \"sonarlint_product\" : \"mediumTests\",\n" +
         "  \"ide_version\" : \"4.5.6\",\n" +
-        "  \"platform\" : \"linux\",\n" +
-        "  \"architecture\" : \"x64\",\n" +
+        "  \"platform\" : \""+ SystemUtils.OS_NAME +"\",\n" +
```
Comment (henryju) [inline medium-tests/src/test/java/mediumtest/TelemetryMediumTests.java:138]:
```suggestion
        "  \"platform\" : \"" + SystemUtils.OS_NAME + "\",\n" +
```

File: rpc-protocol/src/main/java/org/sonarsource/sonarlint/core/rpc/protocol/client/telemetry/TelemetryServerLiveAttributesDto.java [hunk 7]
```diff
@@ -20,6 +20,7 @@
 package org.sonarsource.sonarlint.core.rpc.protocol.client.telemetry;
 
 import java.util.List;
+import javax.annotation.Nullable;
 
 public class TelemetryServerLiveAttributesDto {
```
Comment (henryju) [inline rpc-protocol/src/main/java/org/sonarsource/sonarlint/core/rpc/protocol/client/telemetry/TelemetryServerLiveAttributesDto.java:25]: This class should not be in the RPC protocol, right?

**Review by henryju — CHANGES_REQUESTED** [review]

File: core/src/main/java/org/sonarsource/sonarlint/core/NodeJsHelper.java (hunk 1 above)
Comment (vojtech-suchy-sonarsource) [reply core/src/main/java/org/sonarsource/sonarlint/core/NodeJsHelper.java:67]: I wanted to look educated. But it turned out that using Objects.nonNull is useful when a predicate is needed...I will simplify it.

**Review by henryju — APPROVED** [review]
Well done!

**Pull request merged**